vault/hcp_link/capabilities/meta/meta.go

```diff
@@ -189,3 +191,110 @@ func (h *hcpLinkMetaHandler) ListAuths(ctx context.Context, req *meta.ListAuthsR
 		Auths: auths,
 	}, nil
 }
+
+func (h *hcpLinkMetaHandler) GetClusterStatus(ctx context.Context, req *meta.GetClusterStatusRequest) (*meta.GetClusterStatusResponse, error) {
+	if h.wrappedCore.HAState() != consts.Active {
+		return nil, fmt.Errorf("node not active")
+	}
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return nil, err
+	}
+
+	haEnabled := h.wrappedCore.HAEnabled()
+	haStatus := &meta.HAStatus{
+		Enabled: haEnabled,
+	}
+
+	if haEnabled {
+		leader := &meta.HANode{
+			Hostname: hostname,
+		}
+
+		peers := h.wrappedCore.GetHAPeerNodesCached()
+
+		haNodes := make([]*meta.HANode, len(peers)+1)
+		haNodes[0] = leader
+
+		for i, peerNode := range peers {
+			haNodes[i+1] = &meta.HANode{
+				Hostname: peerNode.Hostname,
+			}
+		}
+
+		haStatus.Nodes = haNodes
+	}
+
+	raftStatus := &meta.RaftStatus{}
+	raftConfig, err := h.wrappedCore.GetRaftConfiguration(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if raftStatus != nil {
+		raftServers := make([]*meta.RaftServer, len(raftConfig.Servers))
+
+		var voterCount uint32
+		for i, srv := range raftConfig.Servers {
+			raftServers[i] = &meta.RaftServer{
+				NodeID:          srv.NodeID,
+				Address:         srv.Address,
+				Voter:           srv.Voter,
+				Leader:          srv.Leader,
+				ProtocolVersion: srv.ProtocolVersion,
+			}
+
+			if srv.Voter {
+				voterCount++
+			}
+		}
+
+		raftStatus.RaftConfiguration = &meta.RaftConfiguration{
+			Servers: raftServers,
+		}
+
+		quorumWarnings := make([]string, 0)
+		if voterCount == 0 {
+			quorumWarnings = append(quorumWarnings, "Only one server node found. Vault is not running in high availability mode.")
+		} else if voterCount%2 == 0 {
+			quorumWarnings = append(quorumWarnings, "Vault should have access to an odd number of voter nodes.")
```

Comment:
Do we also need to add the prefix "Warning:" here?

Reply:
I think that the concept of the warning prefix was from an initial implementation where a single string was built up. Now that they are all surfaced as warnings I will remove the prefix from the "Very large cluster detected" scenario.